`x test tests/run-make/translation` fails when download-rustc is enabled

I tried this code: `./configure --set download-rustc && x test tests/run-make/translation`

I expected to see this happen: All tests pass.

Instead, this happened:
```
rm -rf /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot
mkdir /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot
ln -s /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/* /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot
rm -f /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib
mkdir /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib
ln -s /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/lib/* /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib
rm -f /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib
mkdir /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib
ln -s /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/* /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib
rm -f /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib/src
mkdir /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib/src
ln -s /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/src/* /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/lib/rustlib/src
mkdir -p /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/share/locale/zh-CN/
ln -s /home/jyn/src/rust/tests/run-make/translation/working.ftl /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot/share/locale/zh-CN/basic-translation.ftl
LD_LIBRARY_PATH="/home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation:/home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/lib:/home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage0/lib" '/home/jyn/src/rust/build/x86_64-unknown-linux-gnu/stage2/bin/rustc' --out-dir /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation -L /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation  test.rs --sysroot /home/jyn/src/rust/build/x86_64-unknown-linux-gnu/test/run-make/translation/translation/fakeroot -Ztranslate-lang=zh-CN 2>&1 | "/home/jyn/src/rust/src/etc/cat-and-grep.sh" "this is a test message"
[[[ begin stdout ]]]
error: failed to load fluent bundle: failed to add resource: Attempt to override an existing message: "parse_struct_literal_body_without_path".


[[[ end stdout ]]]
Error: cannot match: this is a test message
------------------------------------------
--- stderr -------------------------------
make: *** [Makefile:51: sysroot] Error 1
------------------------------------------



failures:
    [run-make] tests/run-make/translation

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 308 filtered out; finished in 98.68ms

Some tests failed in compiletest suite=run-make mode=run-make host=x86_64-unknown-linux-gnu target=x86_64-unknown-linux-gnu
```


### Meta
 HEAD is branched from ce1073ba9d894b2e351b2a85fcd39f2c99b78974.

--- a/tests/run-make/translation/Makefile
+++ b/tests/run-make/translation/Makefile
@@ -46,6 +46,8 @@ sysroot: test.rs working.ftl
 	rm -f $(FAKEROOT)/lib/rustlib/src
 	mkdir $(FAKEROOT)/lib/rustlib/src
 	ln -s $(SYSROOT)/lib/rustlib/src/* $(FAKEROOT)/lib/rustlib/src
+	# When download-rustc is enabled, `$(SYSROOT)` will have a share directory. Delete the link to it.
+	rm -f $(FAKEROOT)/share
 	mkdir -p $(FAKEROOT)/share/locale/zh-CN/
 	ln -s $(CURDIR)/working.ftl $(FAKEROOT)/share/locale/zh-CN/basic-translation.ftl
 	$(RUSTC) $< --sysroot $(FAKEROOT) -Ztranslate-lang=zh-CN 2>&1 | $(CGREP) "this is a test message"

